components/device-mgt/org.wso2.carbon.device.mgt.core/src/test/java/org/wso2/carbon/device/mgt/core/notification/mgt/NotificationManagementServiceImplTests.java

```diff
@@ -98,13 +104,66 @@ private RegistryService getRegistryService() throws RegistryException {
 
     @Test(description = "Add notifications using addNotification method and check whether it returns true.")
     public void addNotification() throws Exception {
-        for (int i = 0; i < NO_OF_DEVICES; i++) {
+        Notification notification = null;
+        for (int i = 1; i <= NO_OF_DEVICES; i++) {
             DeviceIdentifier testDeviceIdentifier = new DeviceIdentifier(DEVICE_ID_PREFIX + i, DEVICE_TYPE);
-            Notification notification = TestDataHolder.getNotification(i, "CHECKED",
-                    testDeviceIdentifier.toString(), DEVICE_ID_PREFIX + i, 1, DEVICE_TYPE);
-            NotificationManagementServiceImpl notificationManagementService = new NotificationManagementServiceImpl();
+            notification = TestDataHolder.getNotification(i, Notification.Status.NEW.toString(),
+                    testDeviceIdentifier.toString(), TEST_NOTIFICATION_DESCRIPTION, DEVICE_ID_PREFIX + i,
+                    NOTIFICATION_OPERATION_ID, DEVICE_TYPE);
             Assert.assertTrue(notificationManagementService.addNotification(testDeviceIdentifier, notification));
         }
+        try {
+            notificationManagementService.addNotification(new DeviceIdentifier(DEVICE_ID_PREFIX + 123,
+                    DEVICE_TYPE), notification);
+            Assert.fail();
+        } catch (EntityDoesNotExistException ignored) {
+        }
+    }
+
+    @Test(dependsOnMethods = "addNotification", description = "This tests the updateNotification Method" +
+            " and check whether it returns true ( got updated )")
+    public void updateNotification() throws Exception {
```

Comment:
Why we are throwing generic exception? Can we through specific exception only ? Please correct in other places as well